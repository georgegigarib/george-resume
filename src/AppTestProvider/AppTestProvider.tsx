import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
