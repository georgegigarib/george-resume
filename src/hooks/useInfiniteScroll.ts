import { useEffect, useRef } from 'react'

const useInfiniteScroll = (containerId: string) => {
  const containerRef = useRef<HTMLDivElement>(null)

  useEffect(() => {
    const container = containerRef.current
    if (!container) return

    let animationFrame: number
    let scrollPosition = 0

    try {
      // Duplicamos el contenido para permitir un scroll infinito
      const originalContent = container.innerHTML
      container.innerHTML += originalContent

      // Calculamos el ancho total para ajustar el desplazamiento
      const scrollWidth = container.scrollWidth / 2

      const scrollStep = () => {
        scrollPosition += 0.5 // Ajusta la velocidad del desplazamiento aquÃ­
        if (scrollPosition >= scrollWidth) {
          scrollPosition = 0
        }
        container.scrollLeft = scrollPosition
        animationFrame = requestAnimationFrame(scrollStep)
      }

      scrollStep()
    } catch (error) {
      console.error('Error in useInfiniteScroll:', error)
    }

    // Limpieza al desmontar
    return () => {
      if (animationFrame) cancelAnimationFrame(animationFrame)
    }
  }, [containerId])

  return containerRef
}

export default useInfiniteScroll
