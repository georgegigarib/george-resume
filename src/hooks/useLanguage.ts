import { useTranslation } from 'react-i18next';

export const useLanguage = () => {
  const { i18n } = useTranslation();

  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng);
    localStorage.setItem('language', lng);
  };

  const loadLanguageFromStorage = () => {
    const savedLanguage = localStorage.getItem('language') || 'en';
    i18n.changeLanguage(savedLanguage);
  };

  return { changeLanguage, loadLanguageFromStorage };
};
