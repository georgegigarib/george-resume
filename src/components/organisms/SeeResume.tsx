import React from 'react'
import ResumePDF from '@/assets/documents/George Gil Garib Resume.pdf'
import CVPDF from '@/assets/documents/George Gil Garib CV.pdf'
import resumeIcon from '@/assets/icons/resume-icon.svg'
import { useTranslation } from 'react-i18next'
import { useLanguage } from '@/hooks/useLanguage'
import { useIsMobile } from '@/hooks/useIsMobile'

export default function SeeResume(): React.ReactElement {
  const isMobile = useIsMobile()
  const { t } = useTranslation()
  const { currentLanguage } = useLanguage()

  const pdfMap: Record<string, string> = {
    en: ResumePDF,
    es: CVPDF
  }

  const currentPdf = pdfMap[currentLanguage] || ResumePDF

  return (
    <a
      href={currentPdf}
      target="_blank"
      className={`flex flex-col gap-1 items-center justify-center text-center flex-wrap bg-blue-500 text-white rounded-xl hover:bg-blue-600 transition-all h-full`}
      style={{ boxShadow: '0 4px 12px rgba(0, 0, 0, 0.1)' }}
    >
      <img
        src={resumeIcon}
        alt="Resume Icon"
        data-testid="resume-icon"
        width={isMobile ? '40' : '80'}
        height={isMobile ? '40' : '80'}
        className=""
      />

      <p
        className={`font-bold ${isMobile ? 'text-[0.7em]' : 'text-[1.0em]'}`}
        data-testid="resume-button-text"
      >
        {t('dev.resume.seeResume')}
      </p>
    </a>
  )
}
