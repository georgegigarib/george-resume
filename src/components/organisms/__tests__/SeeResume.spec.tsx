import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi, Mock } from 'vitest';
import SeeResume from '@/components/organisms/SeeResume';
import AppTestProvider from '@/AppTestProvider/AppTestProvider';
import { useIsMobile } from '@/hooks/useIsMobile';
import { useLanguage } from '@/hooks/useLanguage';
import '@testing-library/jest-dom';

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: vi.fn(),
}));

vi.mock('@/hooks/useLanguage', () => ({
  useLanguage: vi.fn(),
}));

describe('SeeResume Component', () => {
  const renderComponent = () =>
    render(
      <AppTestProvider>
        <SeeResume />
      </AppTestProvider>
    );

  it('renders correctly for mobile', () => {
    (useIsMobile as Mock).mockReturnValue(true);

    (useLanguage as Mock).mockReturnValue({ currentLanguage: 'en' });

    const { asFragment } = renderComponent();

    const resumeText = screen.getByTestId('resume-button-text') as HTMLInputElement;

    expect(resumeText.textContent).toBe('See resume');
    expect(screen.getByRole('link')).toHaveAttribute('href', expect.stringContaining('Resume.pdf'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly for desktop', () => {
    (useIsMobile as Mock).mockReturnValue(false);

    (useLanguage as Mock).mockReturnValue({ currentLanguage: 'es' });

    const { asFragment } = renderComponent();

    const resumeIcon = screen.getByAltText('Resume Icon');
    const resumeText = screen.getByTestId('resume-button-text') as HTMLInputElement;

    expect(resumeIcon).toHaveAttribute('width', '80');
    expect(resumeIcon).toHaveAttribute('height', '80');

    expect(resumeText).toBeInTheDocument();
    expect(screen.getByRole('link')).toHaveAttribute('href', expect.stringContaining('CV.pdf'));
    expect(asFragment()).toMatchSnapshot();
  });
});
