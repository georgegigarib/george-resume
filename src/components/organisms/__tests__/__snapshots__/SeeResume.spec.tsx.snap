// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SeeResume Component > renders correctly for desktop 1`] = `
<DocumentFragment>
  <a
    class="flex flex-col gap-1 items-center justify-center text-center flex-wrap bg-blue-500 text-white rounded-lg hover:bg-blue-600 transition-all h-full"
    href="/src/assets/documents/George%20Gil%20Garib%20CV.pdf"
    style="box-shadow: 0 4px 12px rgba(0, 0, 0, 0.1);"
    target="_blank"
  >
    <img
      alt="Resume Icon"
      class=""
      data-testid="resume-icon"
      height="80"
      src="/src/assets/icons/resume-icon.svg"
      width="80"
    />
    <p
      class="font-bold text-[1.0em]"
      data-testid="resume-button-text"
    >
      See resume
    </p>
  </a>
</DocumentFragment>
`;

exports[`SeeResume Component > renders correctly for mobile 1`] = `
<DocumentFragment>
  <a
    class="flex flex-col gap-1 items-center justify-center text-center flex-wrap bg-blue-500 text-white rounded-lg hover:bg-blue-600 transition-all h-full"
    href="/src/assets/documents/George%20Gil%20Garib%20Resume.pdf"
    style="box-shadow: 0 4px 12px rgba(0, 0, 0, 0.1);"
    target="_blank"
  >
    <img
      alt="Resume Icon"
      class=""
      data-testid="resume-icon"
      height="40"
      src="/src/assets/icons/resume-icon.svg"
      width="40"
    />
    <p
      class="font-bold text-[0.7em]"
      data-testid="resume-button-text"
    >
      See resume
    </p>
  </a>
</DocumentFragment>
`;
